recipe/emailverification/utils.go

```diff
@@ -6,25 +6,34 @@ import (
 )
 
 func ValidateAndNormaliseUserInput(appInfo supertokens.NormalisedAppinfo, config schema.TypeInput) schema.TypeNormalisedInput {
-	getEmailVerificationURL := *config.GetEmailVerificationURL
+	var typeNormalisedInput schema.TypeNormalisedInput
+	typeNormalisedInput.GetEmailVerificationURL = *config.GetEmailVerificationURL
 	if config.GetEmailVerificationURL == nil {
-		getEmailVerificationURL = GetEmailVerificationURL(appInfo)
+		typeNormalisedInput.GetEmailVerificationURL = GetEmailVerificationURL(appInfo)
 	}
-	createAndSendCustomEmail := *config.CreateAndSendCustomEmail
+	typeNormalisedInput.CreateAndSendCustomEmail = *config.CreateAndSendCustomEmail
 	if config.CreateAndSendCustomEmail == nil {
-		createAndSendCustomEmail = CreateAndSendCustomEmail(appInfo)
+		typeNormalisedInput.CreateAndSendCustomEmail = CreateAndSendCustomEmail(appInfo)
 	}
 
-	override := schema.Override{
-		Functions: config.Override.Functions,
-		APIs:      config.Override.APIs,
-	}
+	if config.Override != nil {
+		if config.Override.Functions != nil {
+			typeNormalisedInput.Override.Functions = *config.Override.Functions
+		} else {
+			typeNormalisedInput.Override.Functions = func(originalImplementation schema.RecipeInterface) schema.RecipeInterface {
+				return originalImplementation
+			}
+		}
+		if config.Override.APIs != nil {
+			typeNormalisedInput.Override.APIs = *config.Override.APIs
+		} else {
+			typeNormalisedInput.Override.APIs = func(originalImplementation schema.APIInterface) schema.APIInterface {
+				return originalImplementation
+			}
+		}
```

Comment:
- Else? You need to provide a default implementaiton too...
- I think you are using Golang in a way that's actually not giving you all the type safety rules.. Like doing `typeNormalisedInput.xyz = ....` as opposed to `typeNormalisedInput = {...}`.